improver/utilities/neighbourhood_tools.py

```diff
@@ -57,7 +57,10 @@ def rolling_window(input_array, shape, writeable=False):
     """
     num_window_dims = len(shape)
     num_arr_dims = len(input_array.shape)
-    assert num_arr_dims >= num_window_dims
+    if num_arr_dims < num_window_dims:
```

Comment:
As you've kind of shown in your unit test, the argument description above `A 2-D array padded with nans for half the neighbourhood size.` isn't correct as there's no requirement for the input array to be padded with nans. Perhaps something like `2D array from which rolling window neighbourhoods will be created.`?

Reply:
Done. Actually it seems there is no requirement for the array to be 2D so I removed that part.